peps/pep-9999.rst

```diff
@@ -0,0 +1,856 @@
+
+.. list-table::
+    :header-rows: 1
+    :widths: 20 25 40 15
+
+    * - Field
+      - URL
+      - Description
+      - Required
+    * - ``schema``
+      - ``AnyURL``
+      - URL to the mapping for this ecosystem
+      - True
+    * - ``package_managers``
+      - ``list[string]``
+      - List of package manager names known in this ecosystem
+      - True
+
+Mappings schema
+^^^^^^^^^^^^^^^
+
+The mappings are specified by the following
+`JSON Schema <https://github.com/jaimergp/external-metadata-mappings/blob/main/schemas/external-mapping.schema.json>`__:
+
+``$schema``
+~~~~~~~~~~~
+
+.. list-table::
+    :widths: 25 75
+
+    * - Title
+      - $Schema
+    * - Type
+      - ``string``
+    * - Description
+      - URL of the mappings schema in use for the document.
+    * - Required
+      - False
+
+``schema_version``
+~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+    :widths: 25 75
+
+    * - Title
+      - Schema Version
+    * - Type
+      - ``integer``
+    * - Required
+      - False
+
+``name``
+~~~~~~~~
+
+.. list-table::
+    :widths: 25 75
+
+    * - Title
+      - Name
+    * - Type
+      - ``string``
+    * - Description
+      - Name of the schema
+    * - Required
+      - True
+
+``description``
+~~~~~~~~~~~~~~~
+
+.. list-table::
+    :widths: 25 75
+
+    * - Title
+      - Description
+    * - Type
+      - ``string | None``
+    * - Description
+      - Free-form field to add information this mapping. Allows
+        Markdown.
+    * - Required
+      - False
+
+``mappings``
+~~~~~~~~~~~~
+
+.. list-table::
+    :widths: 25 75
+
+    * - Title
+      - Mappings
+    * - Type
+      - ``array``
+    * - Description
+      - List of PURL-to-specs mappings.
+    * - Required
+      - True
+
+Each entry in this list is defined as:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 20 25 40 15
+
+    * - Field
+      - Type
+      - Description
+      - Required
+    * - ``id``
+      - ``PURLField`` (``string`` matching ``^(pkg:|virtual:).*``)
+      - PURL-like identifier, as provided in the central registry
+      - True
+    * - ``description``
+      - ``string``
+      - Free-form field to add some details about the package. Allows Markdown.
+      - False
+    * - ``urls``
+      - ``AnyUrl | list[AnyUrl] | dict[NonEmptyString, AnyUrl]``
+      - Hyperlinks to web locations that provide more information about the definition.
+      - False
+    * - ``specs``
+      - ``string | list[string] | dict[Literal['build', 'host', 'run'], string | list[string]]``
+      - Ecosystem-specific identifiers for this package. The full form is a dictionary
+        that maps the categories ``build``, ``host`` and ``run`` to their corresponding
+        package identifiers. As a shorthand, a single string or a list of strings can be
+        provided, in which case will be used to populate the three categories identically.
+      - Either ``specs`` or ``specs_from`` MUST be present.
+    * - ``specs_from``
```

Comment:
Not sure if this is necessary. Folks can just duplicate existing entries. In most cases this should be an automated process anyway. I'm assuming mapping maintainers will add the relevant PURLs to their package metadata, so these mappings will be created by iterating over the package metadata and extracting the PURL(s), creating one entry per PURL.

Reply:
Simpler might be better in this case I suspect. Okay to leave as is for now though, it doesn't seem like the most critical thing at the moment.